src/main/java/uk/gov/hmcts/authorisation/generators/ServiceAuthTokenGenerator.java

```diff
@@ -0,0 +1,30 @@
+package uk.gov.hmcts.authorisation.generators;
+
+import com.warrenstrange.googleauth.GoogleAuthenticator;
+import uk.gov.hmcts.authorisation.ServiceAuthorisationApi;
+
+import static java.lang.Integer.valueOf;
+
+public class ServiceAuthTokenGenerator implements AuthTokenGenerator {
+
+    private final String secret;
+    private final String microService;
+    private final ServiceAuthorisationApi serviceAuthorisationApi;
+
+    public ServiceAuthTokenGenerator(
+            final String secret,
+            final String microService,
+            final ServiceAuthorisationApi serviceAuthorisationApi
+    ) {
+        this.secret = secret;
+        this.microService = microService;
+        this.serviceAuthorisationApi = serviceAuthorisationApi;
+    }
+
+    @Override
+    public String generate() {
+        final GoogleAuthenticator googleAuthenticator = new GoogleAuthenticator();
+        final String oneTimePassword = valueOf(googleAuthenticator.getTotpPassword(secret)).toString();
```

Comment:
this convert `int` -> `Integer` -> `String`

Use either `String.valueOf(i)` or `Integer.toString(i)` instead.

Reply:
also, does it not need to be padded with leading zeros?

Reply:
reply it doesn't need, it works as it